lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -271,10 +271,6 @@ def parse_distribution_file_SUSE(self, name, data, path, collected_facts):
                     else:
                         release = "0"  # no minor number, so it is the first release
                     suse_facts['distribution_release'] = release
-                # Starting with SLES4SAP12 SP3 NAME reports 'SLES' instead of 'SLES_SAP'
-                # According to SuSe Support (SR101182877871) we should use the CPE_NAME to detect SLES4SAP
-                if re.search("^CPE_NAME=.*sles_sap.*$", line):
```

Comment:
Agreed with @Akasurde below about backwards compat. Is there any reason to remove this? Would it be better to keep this here, and simply add the below as another way to set `suse_facts['distribution'] = 'SLES_SAP'` if it didn't get set here for some reason?